pkg/plugins/plugins.go

```diff
@@ -41,26 +41,33 @@ var (
 )
 
 type PluginScanner struct {
-	pluginPath           string
-	errors               []error
-	backendPluginManager backendplugin.Manager
-	cfg                  *setting.Cfg
-	requireSigned        bool
-	log                  log.Logger
-	plugins              map[string]*PluginBase
+	pluginPath                    string
+	errors                        []error
+	backendPluginManager          backendplugin.Manager
+	cfg                           *setting.Cfg
+	requireSigned                 bool
+	log                           log.Logger
+	plugins                       map[string]*PluginBase
+	allowUnsignedPluginsCondition func(plugin *PluginBase) bool
 }
 
 type PluginManager struct {
-	BackendPluginManager backendplugin.Manager `inject:""`
-	Cfg                  *setting.Cfg          `inject:""`
-	log                  log.Logger
-	scanningErrors       []error
+	BackendPluginManager          backendplugin.Manager `inject:""`
+	Cfg                           *setting.Cfg          `inject:""`
+	log                           log.Logger
+	scanningErrors                []error
+	allowUnsignedPluginsCondition func(plugin *PluginBase) bool
 }
 
 func init() {
 	registry.RegisterService(&PluginManager{})
 }
 
+// ReplaceAllowUnsignedPluginCondition has to be called before PluginManager.Init() has run
```

Comment:
This comment might not age very well as Grafana gets more dynamic during runtime. I've proposed an alternative:

```suggestion
// ReplaceAllowUnsignedPluginCondition changes the policy for allowing unsigned plugins. Signature validation only runs when plugins are starting
// and running plugins will not be terminated if they violate the new policy.
// Run this before PluginManager.Init to be sure policy is applied to all loaded plugins.
```